### PR TITLE
Fix crashing issues with RSSI

### LINKED ISSUE #121: WiFi.RSSI() causing lock up on MKR1000
_Moved from https://github.com/arduino/Arduino/issues/5725_ @tonywhitfort 

The following code demonstrates WiFi.RSSI() causing the processor to lock up.
In this example the lockup is demonstrated by the LED which stops flashing after a few seconds or sometimes a bit longer after a restart.
I am using MKR1000 with the WiFi101 library.

I have found that not calling WiFi.RSSI() for 20secs after WiFi.begin() seems to fix the problem
Shorter delays cause varying degrees of unreliability, the shorter the worse.
See the code comments for more detail

Sorry for the lack of code formatting - I tried using the insert code feature but it didn't seem to have the desired affect - not in the Preview tab away. What am I missing?

```C++
#include <WiFi101.h>

char ssid[] = "xx"; 
char pass[] = "xx";   

#define SECS  2
// for SECS = 2  - very unreliable - locks up after a few seconds more often than not
// for SECS = 10 - more reliable   - locks up on every 5th to 10th restart
// for SECS = 20 - seems reliable

void setup() 
{ 
  pinMode(LED_BUILTIN, OUTPUT);
  
  int status = WL_IDLE_STATUS;  
  while (status != WL_CONNECTED) {  
    status = WiFi.begin(ssid, pass);
  }
}

void loop() 
{
  static int x = 0;

  if (millis()/1000 < SECS) { 
    digitalWrite(LED_BUILTIN, x++ % 5 ? HIGH : LOW); // slow flash   
  } else {   
    WiFi.RSSI();
    digitalWrite(LED_BUILTIN, x++ % 2 ? HIGH : LOW); // fast flash to identify when delay finished
  }
  delay(100);
}
```

### PR DESCRIPTION
**Pull request by sandeepmistry**
Fixes #121.

cc/ @facchinm @agdl @akash73

### CODE REVIEW

File: src/WiFi.cpp [hunk 1]
```diff
@@ -259,6 +259,7 @@ int WiFiClass::init()
 	_gateway = 0;
 	_dhcp = 1;
 	_resolve = 0;
+	_remoteMacAddress = 0;
```
Comment (facchinm) [inline src/WiFi.cpp]: Is this change related with RSSI issue?

**Comment by tonywhitfort** [comment]
OK, that works brilliantly! 
Thanks sandeepmistry your the man!
I spent a whole day scratching my head over that one - nice to have that behind me.

**Comment by sandeepmistry** [comment]
@tonywhitfort thanks for testing the change out, and providing an excellent issue report to let us know about the problem!

There's a new v0.11.2 release that contains the fix.

**Comment by tonywhitfort** [comment]
I have updated my library to v0.11.2 and all good
Thanks for the quick fix

**Pull request merged**